packages/xstate-react/src/useActor.ts

```diff
@@ -0,0 +1,69 @@
+import { useState, useEffect, useMemo, useRef, useCallback } from 'react';
+import { ActorRef, Sender, ActorRefLike } from './types';
+import { EventObject, Interpreter, Actor } from 'xstate';
+import { fromService } from './useService';
+
+function resolveActor(
+  actorLike:
+    | ActorRef<any, any>
+    | ActorRefLike<any, any>
+    | Interpreter<any, any>
+    | Actor
+): ActorRef<any, any> & { deferred?: boolean } {
+  if (actorLike instanceof Interpreter) {
+    return fromService(actorLike);
+  }
+
+  if (!('current' in actorLike)) {
+    return {
+      stop: () => {
+        /* do nothing */
+      },
+      ...actorLike,
+      subscribe: actorLike.subscribe as any, // TODO: fix (subscribe will work, just improperly typed)
+      name: actorLike.id!,
+      current: actorLike.state!
+    };
+  }
+
+  return actorLike;
+}
+
+export function useActor<TEvent extends EventObject, TEmitted = any>(
```

Comment:
some tests would be great for this, especially given that `resolveActor` accepts quite a few variations of Actors and it's quite hard to see right now how it works with all of those variations

Reply:
I want to merge this in first and flesh out how it will work with the current V5 implementation - will write tests there.